portfolio/src/main/java/com/google/sps/servlets/LoginStatusServlet.java

```diff
@@ -0,0 +1,50 @@
+package com.google.sps.servlets;
+
+import com.google.appengine.api.datastore.DatastoreService;
+import com.google.appengine.api.datastore.DatastoreServiceFactory;
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.FetchOptions;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+import com.google.appengine.api.datastore.Query.SortDirection;
+import com.google.appengine.api.users.UserService;
+import com.google.appengine.api.users.UserServiceFactory;
+import com.google.gson.Gson;
+import com.google.sps.data.Comment;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** Servlet that returns the login status of the user. */
+@WebServlet("/login-status")
+public class LoginStatusServlet extends HttpServlet {
+  @Override
+  public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    // Get the current login status.
+    LoginStatus status;
+    if (UserServiceFactory.getUserService().isUserLoggedIn()) {
+      status = new LoginStatus(true);
+    } else {
+      status = new LoginStatus(false);
+    }
+
+    // Send the JSON as the response
+    response.setContentType("application/json;");
+    response.getWriter().println(convertToJson(status));
+  }
+
+  private String convertToJson(LoginStatus status) {
+    return new Gson().toJson(status);
+  }
+
+  private class LoginStatus {
```

Comment:
As you only have 2 kinds of status: true and false. I think you can make this class Singleton. (Please ignore this comment if you have more to add very soon)

You can refer to this one:
https://www.geeksforgeeks.org/java-singleton-design-pattern-practices-examples/

I think what you need is: ( Use a better name if you can. I am not good at naming ;-) )
private static final LOGIN_STATUS = new LoginStatus(true);
private static final NON_LOGIN_STATUS = new LoginStatus(false);

And make the constructor private, and provide getters for the 2 STATUS

Reply:
Thank you, I am looking into it. I will either add in this PR or the next one [2/X].

Reply:
I have converted it into a Singleton class. This forced me to declare the class as static due to  the class being an inner class but I think this is OK. Reference: https://stackoverflow.com/questions/6440010/modifier-static-is-only-allowed-in-constant-variable-declarations